### DISCUSSION TITLE
How to check if the user has reposted something

### DISCUSSION THREAD

**Discussion by GreenFootballs** [bluesky-social/atproto#1734] [How to do it?] [answered]
The client I'm building needs to do and undo reposts, but how do I get the `rkey` value for the `deleteRecord` call? I checked by undoing a repost in the Bluesky app and watching Network requests, and the `rkey` passed to `deleteRecord` is not the same as the `rkey` in that post's URL. So how do I find it?

Conversely, how can I tell if the user has reposted a specific post?

**Comment by DavidBuchanan314** [comment, marked as answer]
The response of an `app.bsky.feed.getPostThread` request will include the URIs of the viewer's like and repost records, if present, in the "viewer" field. (thread -> post -> viewer -> repost, for example, being the full json path)

**Reply by GreenFootballs** [reply]
Thanks, that does it.
